### PR TITLE
Add UInt64AddOperator to rocksjava

### PR DESCRIPTION
**Pull request by jcalcote**
Closes https://github.com/facebook/rocksdb/issues/4447

### CODE REVIEW

File: java/CMakeLists.txt [hunk 1]
```diff
@@ -132,6 +132,9 @@ set(NATIVE_JAVA_CLASSES
         org.rocksdb.TtlDB
         org.rocksdb.VectorMemTableConfig
         org.rocksdb.WBWIRocksIterator
+        org.rocksdb.Snapshot
+        org.rocksdb.StringAppendOperator
```
Comment (jcalcote) [inline java/CMakeLists.txt]: oops - these should probably not have been added

Comment (jcalcote) [reply java/CMakeLists.txt]: fixed

**Comment by jcalcote** [comment]
I don't know why AppVeyor build is failing - the log shows the GroupCommitTest failed - but I didn't change anything I'm aware of that is related to that test.

**Comment by sagar0** [comment]
Could you please sign the CLA before we look at this further? 
The appveyor failure is not due to your code. I restarted the appveyor build on this PR, and it passed.

**Comment by jcalcote** [comment]
Signed - thanks!
John

**Comment by lolski** [comment]
This would be a great addition to the library! Is there any timeline for when it is going to get merged?

File: java/src/main/java/org/rocksdb/UInt64AddOperator.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2018, Vlad Balan (vlad.gm@gmail.com).  All rights reserved.
```
Comment (adamretter) [inline java/src/main/java/org/rocksdb/UInt64AddOperator.java]: Please remove this copyright line or change it to:
```
Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
```

Comment (jcalcote) [reply java/src/main/java/org/rocksdb/UInt64AddOperator.java]: will do - I just copied the header from StringAppendOperator.java - would you like me to change that one too?

Comment (adamretter) [reply java/src/main/java/org/rocksdb/UInt64AddOperator.java]: Nope, leave the StringAppendOperator.java one as it is.

Comment (jcalcote) [reply java/src/main/java/org/rocksdb/UInt64AddOperator.java]: done.

**Comment by jcalcote** [comment]
@adamretter - do you ever accept patches for older branches? I need (at this time) to use this patch on the 5.2.fb branch (creating a 5.2.2 release) and would also want that branch updated with a 5.2.2 jar on maven central. If this is way outside what you guys do, no worries - I can clone and build 5.2.2 locally if needed, but wanted to avoid it if I could.

**Comment by ajkr** [comment]
@jcalcote sorry but our policy for RocksDB repo is to only backport bug fixes to older versions. But I guess you can locally add it.

File: java/src/main/java/org/rocksdb/UInt64AddOperator.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+// Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
```
Comment (sagar0) [inline java/src/main/java/org/rocksdb/UInt64AddOperator.java]: You need to have the other license related lines too after this copyright line. So, could you please add back:
```
//  This source code is licensed under both the GPLv2 (found in the
//  COPYING file in the root directory) and Apache 2.0 License
//  (found in the LICENSE.Apache file in the root directory).
```

Comment (sagar0) [reply java/src/main/java/org/rocksdb/UInt64AddOperator.java]: Thanks for taking care of this @ajkr .

**Pull request closed without merging**